casadm/cas_lib.c

```diff
@@ -1866,7 +1866,22 @@ int remove_core(unsigned int cache_id, unsigned int core_id,
 	struct kcas_remove_core cmd;
 
 	/* don't even attempt ioctl if filesystem is mounted */
-	if (core_mounts_detected(cache_id, core_id)) {
+	bool mounts_detected = false;
+	int cmplen = 0;
+	char pattern[80];
+
+	/* verify if specific core is mounted */
+	cmplen = snprintf(pattern, sizeof(pattern), "/dev/cas%d-%d", cache_id, core_id);
+	if (!(mounts_detected = device_mounts_detected(pattern, cmplen))) {
```

Comment:
I would split this into a separate lines with assignment and then check, just for readability.